python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -122,62 +122,53 @@ async def get_chat_message_contents(
         Returns:
             A list of chat message contents.
         """
+        if (
+            settings.function_choice_behavior is None
+            or not settings.function_choice_behavior.auto_invoke_kernel_functions
+        ):
+            return await self._send_chat_request(chat_history, settings)
+
+        self._verify_function_choice_behavior(settings, **kwargs)
+        kernel: Kernel = kwargs.get("kernel")
+        arguments: KernelArguments = kwargs.get("arguments")
+        self._configure_function_choice_behavior(settings, kernel)
+
+        for request_index in range(settings.function_choice_behavior.maximum_auto_invoke_attempts):
+            completions = await self._send_chat_request(chat_history, settings)
+            chat_history.add_message(message=completions[0])
+            function_calls = [item for item in chat_history.messages[-1].items if isinstance(item, FunctionCallContent)]
+            if (fc_count := len(function_calls)) == 0:
+                return completions
+
+            results = await self._process_function_calls(
+                function_calls=function_calls,
+                chat_history=chat_history,
+                kernel=kernel,
+                arguments=arguments,
+                function_call_count=fc_count,
+                request_index=request_index,
+                function_behavior=settings.function_choice_behavior,
+            )
+
+            if any(result.terminate for result in results if result is not None):
+                return completions
+        else:
+            # do a final call without auto function calling
+            return await self._send_chat_request(chat_history, settings)
+
+    async def _send_chat_request(
+        self, chat_history: ChatHistory, settings: AzureAIInferenceChatPromptExecutionSettings
+    ) -> list[ChatMessageContent]:
+        """Send a chat request to the Azure AI Inference service."""
         response: ChatCompletions = await self.client.complete(
-            messages=self._format_chat_history(chat_history),
+            messages=format_chat_history(chat_history),
```

Comment:
(we had a conversation about this in the original PR and I changed my mind on this, sorry) but I would prefer if we use the function self._prepare_chat_history_for_request(chat_history) across services, the return type will be different for this one, which is fine, no need to add the 'override' decorator so that mypy doesn't complain, but doing that will keep things more consistent across services and thereby easier to read and understand what's going on.

Reply:
I am open to this. Though having the same function name could improve readability but having different returned types of the 'same' function may cause confusion at the same time.

Another suggestion is to mark the return type of self._prepare_chat_history_for_request(chat_history) to Any.

Reply:
Let's do that, I think the name being the same throughout helps and having the generic return Any makes it clear that it can be adapted to the situation which is what we want!